### ISSUE TITLE
nativeFunction exception handling

### ISSUE THREAD

**Issue by MiauwMaster** [frida/frida#873]
I am working on writing a fuzzing tool based on frida. I create a new nativeFunction and want to propagate any exceptions that occur. I have created an Process.setExceptionHandler(debuggr) with the debuggr function just set up to print the exception details for now. 

I also have a small test program with has a buffer overflow vulnerability. I am able to trigger this vuln without issue with this code
'''javascript
var fuzzfunc = new NativeFunction(ptr('functionMemoryAddressHere'), 'bool', ['pointer'], ['propagate']);

this.sendtext = Memory.allocUtf8String("AAAAAAAAAAAAAAAAAAAA");
var result = fuzzfunc(this.sendtext);
'''

The problem I'm having is that when the buffer overflow triggers, the exception is NOT caught. the debuggr function is never called and the program just dies.
My frida load form a C file(based off of the frida-core example) and works fine, I've also tried in python, same (lack of) result.
Any help is highly appreciated.

**Comment by bannsec** [comment]
I just banged my head on the exact same issue. I don't think exception handling is implemented for native functions.

**Comment by oleavr** [comment]
@bannsec It is. It will turn the exception into a JS exception that you can try/catch. But you can configure the `NativeFunction` to propagate exceptions instead of stealing them, where the latter is the default (see docs). Then your process-wide exception handler has a chance to handle them. But, if this doesn't work and the process crashes, it means there's a bug on that specific platform, and we should fix that.

**Comment by bannsec** [comment]
Fair enough. Didn't notice that exception doc for native functions. Will try using that and see if it works.

**Comment by bannsec** [comment]
Ok. I can confirm it works. As you mentioned, added try/catch around the excepting nativefunction and it returns the exception.

I will say that I've intermittently been having the nativeContext be empty (0x0 or whatever). I have it creating a traceback from this context, so sometimes i don't get the traceback.

Might be worth placing a note in the Process addexceptionhandler doc that the exception handler might be caught by NativeFunction instead for folks like myself that google search for exception handling in frida and get pointed to that.

**Comment by bannsec** [comment]
Actually, looks like the traceback is only tracing back frida's own library for the crash:

```
([{'type': 'access-violation', 'address': '0x5630b6b0b04e', 'memory': {'operation': 'read', 'address': '0x0'}, 'context': {'pc': '0x5630b6b0b04e', 'sp': '0x7f02c77fc7b0', 'rax': '0x7f02b8001280', 'rcx': '0x7f02b8001288', 'rdx': '0x2', 'rbx': '0x7f02c77fd170', 'rsp': '0x7f02c77fc7b0', 'rbp': '0x7f02c77fd230', 'rsi': '0x2', 'rdi': '0x0', 'r8': '0x0', 'r9': '0x1', 'r10': '0x7f02b80008d0', 'r11': '0x0', 'r12': '0x0', 'r13': '0x2', 'r14': '0x7f02d03b7f78', 'r15': '0x7f02c6efbba0', 'rip': '0x5630b6b0b04e'}, 'nativeContext': '0x0'}, ['0x7f02cd4e6eae', '0x7f02cd4fba19', '0x7f02cd3d033f', '0x7f02cd4fa9e0', '0x7f02cd4fb9db', '0x7f02cd516232', '0x7f02cd500221', '0x7f02cd510cb4', '0x7f02cd4e1eba', '0x7f02cd4d9c44', '0x7f02cd4d72e2', '0x7f02cd463896', '0x7f02cd463ac0', '0x7f02cd463cea', '0x7f02cd4d739f', '0x7f02cd46d169']], [None, None])```

Is there a way to get the backtrace of the excepted thread instead of the backtrace for the frida javascript engine?
